include/http2.hrl

```diff
@@ -85,8 +85,8 @@
 
 -type stream_id() :: non_neg_integer().
 -record(frame_header, {
-    length      :: non_neg_integer(),
-    type        :: frame_type(),
+    length      :: non_neg_integer() | undefined,
```

Comment:
Ah, I see what happens here. This frame header is partially constructed, and it's finished in `h2_frame:to_binary` but it does exist in this intermediate state.

🤔 thinking